xonsh/tools.py

```diff
@@ -1171,7 +1171,7 @@ def backup_file(fname):
     import shutil
     from datetime import datetime
     base, ext = os.path.splitext(fname)
-    newfname = base + '.' + datetime.now().isoformat() + ext
+    newfname = '%s.%s%s' % (base, datetime.now().isoformat().replace(':', '-'), ext)
```

Comment:
To me, I think it is more readable just to use our own formatting string.  Maybe something like the following?
`time.strftime('%Y-%m-%d-%H-%M-%S-%f')`